src/composables/mutations/useUpsertOrgMutation.ts

```diff
@@ -71,20 +116,20 @@ const useUpsertOrgMutation = () => {
   const authStore = useAuthStore();
   const queryClient = useQueryClient();
 
-  return useMutation<OrgData, Error, OrgData, UpsertOrgMutationContext>({
+  return useMutation<OrgType, Error, CreateOrgType, UpsertOrgMutationContext>({
     mutationKey: [ORG_MUTATION_KEY],
-    mutationFn: async (data: OrgData): Promise<OrgData> => {
-      const formattedOrgData = formatOrgData(data);
-      await authStore.roarfirekit.upsertOrg(formattedOrgData);
-      return formattedOrgData;
+    mutationFn: async (data: CreateOrgType): Promise<OrgType> => {
+      const parsed = parseCreateOrgData(data);
+      const response = await authStore.roarfirekit.upsertOrg(parsed);
+      return response.data;
```

Comment:
We should handle the case if parseCreateOrgData throws an error because the mutation function is expected to return a promise. The validation should happen before we pass it to the mutation. 

https://tanstack.com/query/latest/docs/framework/react/reference/useMutation